hpx/traits/action_direct_execution.hpp

```diff
@@ -0,0 +1,29 @@
+//  Copyright (c) 2016 Thomas Heller
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_TRAITS_ACTION_DIRECT_EXECUTION_HPP
+#define HPX_TRAITS_ACTION_DIRECT_EXECUTION_HPP
+
+#include <hpx/util/always_void.hpp>
+
+#include <type_traits>
+
+namespace hpx { namespace traits {
+    template <typename Action, typename Enable = void>
+    struct action_direct_execution
+      : Action::direct_execution
+    {};
+}}
+
+#define HPX_TRAITS_ACTION_DIRECT_EXECUTION(Action, DirectExecution)            \
```

Comment:
The idea for that was to add a kind of tag type to `base_lco` and `base_lco_with_value` to allow direct or non direct action execution based on the actual implementation of the LCO. Implementing that turned out to be a little challenging, so I decided to drop that for now. I am fine with removing this trait.